tests/multicomponent_compressible_projected_density_box.prm

```diff
@@ -0,0 +1,151 @@
+#                    isothermal_bulk_modulus_pressure_derivatives   *
+#                          reference_isothermal_compressibilities)
+# densities[c] = reference_densities*std::pow(f, 1./isothermal_bulk_modulus_pressure_derivatives)
+
+# Which should result in a density of 3822.3642114512877, and as we are only
+# prescribing a constant temperature and pressure, the average density, 
+# as well as the value of the density_field composition should both also be equal to 3822.3642114512877.
+
+set Dimension = 2 
+set End time                               = 1e6
+set Use years in output instead of seconds = true
+set Adiabatic surface temperature          = 1600
+set Surface pressure                       = 5e10
+set Pressure normalization                 = surface
+set Nonlinear solver scheme                = iterated Advection and Stokes
+set Nonlinear solver tolerance             = 1e-12
+set Max nonlinear iterations               = 25
+set Output directory                       = visco_plastic_compressible_box
+
+# This calculates the adiabatic temperature profile down the model and is used
+# by the initial temperature and composition models.
+subsection Adiabatic conditions model
+  subsection Compute profile
+    set Number of points              = 100
+    set Composition reference profile = function
+  end
+end
+
+subsection Solver parameters
+  subsection Stokes solver parameters
+    set Linear solver tolerance = 1e-12
+  end
+end
+
+# The projected density approximation requires prescribed fields
+subsection Compositional fields
+  set Number of fields = 1
+  set Names of fields  = density_field
+  set Compositional field methods = prescribed field
+  set Types of fields             = density
+end
+
+# This is the functionality being tested. The projected density approximation
+# is the preferred compressibility formulation, as demonstrated by
+# the compressibility formulations benchmark (Gassmoeller et al.).
+# The composition statistics postprocessor requires that the single field we have
+# be filled, which ensures that the density we see has been prescribed.
+subsection Formulation
+  set Mass conservation    = projected density field
+  set Temperature equation = real density
+end
+
+subsection Initial composition model
+  set Model name = adiabatic density
+end
+
+# 1x1 m box with a 2x2 grid (which is the minimum)
+# there are no global or adaptive refinement steps
+# so this is the initial and end resolution.
+subsection Geometry model
+  set Model name = box
+
+  subsection Box
+    set Y extent = 1
+    set X extent = 1
+    set X repetitions = 1
+    set Y repetitions = 1
+  end
+end
+
+# We only want to evaluate the density due to our applied surface
+# pressure, so we disable gravity.
+subsection Gravity model
+  set Model name = vertical
+
+  subsection Vertical
+    set Magnitude = 0.0
+  end
+end
+
+############### Boundary conditions
+# We fix the temperature on all boundaries, and as our model is so small this
```

Comment:
I would consider modifying this section so that the boundary temperature is derived from the initial conditions.

```
subsection Boundary temperature model
  set Fixed temperature boundary indicators = left, right, top, bottom
  set List of model. names = initial temperature
end
```